# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
'''
# I have chosen the **Refresh / Content Opportunity Scoring** lane.

# The goal of this project is to identify which webpages should be refreshed first to improve their search performance. Many websites contain thousands of pages, making it impossible for content teams to review every page manually. A data-driven ranking system can help prioritize the pages that are most likely to benefit from an update.

# I chose this lane because it focuses on solving a real business problem rather than simply building a machine learning model. The final output can directly help SEO and content teams decide where to spend their time and resources.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
'''

### Question

 Which webpages should be reviewed first for refresh, expansion, protection, pruning, or monitoring based on observable search and content signals?

### Decision

 The decision is determining which pages should be prioritized for review instead of reviewing every page equally.

### Action

 The SEO or content team will review the highest-ranked pages first and decide whether they should be refreshed, expanded, protected, monitored, or pruned.

### Unit of Analysis

One webpage (one content page).

### Output

 A ranked opportunity score for every page along with suggested actions, confidence level, and reason codes.

### Cost of a Wrong Recommendation

If the system incorrectly prioritizes pages that have little opportunity, valuable time and resources will be wasted.

If important pages are not recommended for review, the organization may lose search visibility, traffic, and business opportunities.

Therefore, incorrect recommendations reduce the effectiveness of limited SEO resources.

### Why Machine Learning?

Large websites may contain thousands of webpages with many observable search and content signals.

Machine learning can identify useful patterns within these signals and rank pages according to their potential opportunity.

The purpose is to support human decision-making by producing a ranked review queue instead of making automatic decisions.
'''

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/acecod3z/Flyrankinternship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print(df.shape)
df.head()

print("Shape:", df.shape)
print("Unique clients:", df["client_id"].nunique())
print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

(30000, 44)
Shape: (30000, 44)
Unique clients: 32

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Quick Look at the Data

I loaded the starter dataset and explored some basic statistics to understand whether this lane is suitable.

- The dataset contains **30,000 content pages** described by **44 features**.
- The data comes from **32 unique anonymized clients**, providing a variety of websites and content.
- The trend distribution shows that **16,262 pages are currently classified as "down"**, while **5,962 are stable** and **4,388 are trending up**. This indicates that a large portion of the dataset consists of pages that may require review, making the Refresh / Content Opportunity Scoring lane a meaningful problem to investigate.

These observations suggest that there is enough scale and variation in the dataset to build a ranked review queue that helps prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I Can and Can't Claim

This project is designed to support content review decisions, not to guarantee SEO improvements. A high opportunity score does not prove that refreshing a page will increase traffic or rankings. The recommendations are based on observable historical search and content signals available in the starter dataset.

The ranked review queue should be used as a decision-support tool to help prioritize pages for human review. Final actions such as refreshing, expanding, protecting, pruning, or monitoring content should still involve human judgment and additional context that is not available in the dataset.

This analysis identifies patterns and associations in the data. It does not establish cause-and-effect relationships, and the results are limited to the information contained in the anonymized starter dataset.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.